documentation/src/test/java/org/hibernate/search/documentation/mapper/orm/schema/management/HibernateOrmSchemaManagerIT.java

```diff
@@ -84,6 +96,57 @@ public void selectType() {
 		} );
 	}
 
+	@Test
+	public void walkingTheSchema() {
+		with( entityManagerFactory ).runNoTransaction( entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+			List<String> indexNames = new ArrayList<>();
+			// tag::walking-the-schema[]
+			SearchSchemaManager schemaManager = searchSession.schemaManager(); // <1>
+			schemaManager.exportSchema(
+					new SearchSchemaCollector() { // <2>
+						@Override
+						public void indexSchema(IndexSchemaExport export) {
+							String indexName = export.indexName(); // <3>
+							// perform any other actions with an index schema export
+							// end::walking-the-schema[]
+							indexNames.add( indexName );
+							// tag::walking-the-schema[]
+						}
+					}
+			);
+			// end::walking-the-schema[]
+			assertThat( indexNames ).containsOnly( Book.class.getSimpleName(), Author.class.getSimpleName() );
+		} );
+	}
+
+	@Test
+	public void exportSchemaToFiles() throws IOException {
+		with( entityManagerFactory ).runNoTransaction( entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+			Path targetDirectory = temporaryFolder.newFolder().toPath();
+
+			// tag::schema-export[]
+			SearchSchemaManager schemaManager = searchSession.schemaManager(); // <1>
+			schemaManager.exportSchema( SearchSchemaCollectors.indexSchemaToFile( targetDirectory ) ); // <2>
+			// end::schema-export[]
+
+			String bookIndex = Files.readString(
+					targetDirectory.resolve( "backend" )
+							.resolve( "indexes" )
+							.resolve( Book.class.getSimpleName() )
+							.resolve( "index.json" ) );
```

Comment:
Yeah 😃 I have mixed feelings about that one 😄 
I do like the idea of having multiple files or files specific to the backend.. But on the other hand, I had a "problem" with that `toDirectory` method being sort of out of place... what if we'd needed to export to some other place, maybe through HTTP or something? Would we be adding more `toSmth()` methods?
I'd prefer to remove the `write` or `toDirectory` methods from this export interface and have the actual exporting logic live in the collector implementation. But that would mean that the collector from the engine would need to delegate to the collectors from backends, which leads to probably otherwise unnecessary complexity 😄 
Anywaaay, if you are ok with `toDirectory(..)` I'll make the changes 😃

Reply:
> what if we'd needed to export to some other place, maybe through HTTP or something

We're free to reject features that we don't think make sense, and in this case I would.

But if you really don't want to rely on the filesystem at this level, I'd say you could use something like `.toFiles(BiConsumer<String, OutputStream>)`.